docs/source/refs/reference_architecture/index.rst

```diff
@@ -0,0 +1,392 @@
+their own environments. For more technical information about the two workflows, please see the
+`documentation <https://isaac-sim.github.io/IsaacLab/main/source/overview/core-concepts/task_workflows.html>`__.
+
+
+In addition to designing the RL task, you will need to design your agent’s model, the neural
+network policy and value function. To train the RL agent to solve the task, you need to define
+the hyperparameters such as number of epochs, learning rate, etc. for training and the
+policy/value model architecture. This is defined in the training configuration file specific
+to the RL library you want to use. Examples are created under the agent's folder in each task directory.
+See an example of `RSL-RL <https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/locomotion/velocity/config/anymal_b/agents/rsl_rl_ppo_cfg.py>`__ for Anymal-B.
+
+
+**Component 4 - Register with Gymnasium**
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The next step is to register the environments with the gymnasium registry.
+Registration is a way to make the environment accessible and reusable across different
+RL algorithms and experiments. This is common in the RL community. Follow the tutorial on
+`Registering an Environment <https://isaac-sim.github.io/IsaacLab/main/source/tutorials/03_envs/register_rl_env_gym.html>`__ to learn more about how to register in your own environment.
+
+**Component 5 - Environment Wrapping**
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+In running your RL task, you might want to change the behavior of your environment without
+changing the environment itself. For example, you might want to create functions to modify
+observations or rewards, record videos, or enforce time limits. Isaac Lab utilizes the API
+available in the `gymnasium.Wrapper <https://gymnasium.farama.org/api/wrappers/#gymnasium.Wrapper>`__ class to create interfaces to the simulated environments.
+
+Some wrappers include:
+
+* `Video Wrappers <https://isaac-sim.github.io/IsaacLab/main/source/how-to/wrap_rl_env.html#wrapper-for-recording-videos>`__
+* `RL Libraries Wrappers <https://isaac-sim.github.io/IsaacLab/main/source/how-to/wrap_rl_env.html#wrapper-for-learning-frameworks>`__
+
+Most RL libraries expect their own variation of an environment interface. This means the
+data types needed by each library differs. Isaac Lab provides its own wrappers to convert
+the environment into the expected interface by the RL library a user wants to use. These are
+specified in the `Isaac Lab utils wrapper module <https://isaac-sim.github.io/IsaacLab/main/source/api/lab_tasks/omni.isaac.lab_tasks.utils.wrappers.html#module-omni.isaac.lab_tasks.utils.wrappers>`__.
+
+See the `full list <https://gymnasium.farama.org/api/wrappers/#gymnasium.Wrapper>`__ of other wrappers APIs.. For more information on how these wrappers work,
```

Comment:
is this the correct link?

also typo with ".."

Reply:
Link changed and ".." corrected